src/translations/maps/queuedPhysicalIndexToValueMap.js

```diff
@@ -0,0 +1,103 @@
+import IndexMap from './indexMap';
+import { getListWithRemovedItems, getListWithInsertedItems } from './utils/physicallyIndexed';
+import { getDecreasedIndexes, getIncreasedIndexes } from './utils/actionsOnIndexes';
+import { isFunction } from '../../helpers/function';
+
+/**
+ * Map for storing mappings from an physical index to a value. Some values are stored in a certain order.
+ *
+ * Does not update stored values on remove/add row or column action. Otherwise, queue of indexes related to ordered
+ * values is updated after such changes.
+ */
+class QueuedPhysicalIndexToValueMap extends IndexMap {
```

Comment:
I'm not convinced about the match of the word "Queued". The Queue in the definition is a structure that accepts values on the one side and removes the values on the other side - nothing happens here. Moreover IMO "queue" can suggest that something is collected for further processing (maybe asynchronous). 

I think that the better word could be "Ordered". It doesn't tell anything about the process method and still describes the meaning that the values will be accessible in the same order as they were put into. 

@wszymanski @wojciechczerniak what do you guys think about it?

Reply:
Leaving aside the fact that queue is also a term used for describing things beyond programming (and for me, it may be still useful for naming abstract beings; I use sometimes, for example, the phrase `list` for naming things not necessarily implementing the list in the sense of programming) I would like to pay attention to the fact there are different types of queue 🙂 

In programming, we also distinguish for example **priority queue**. We add elements to an array using priority starting from elements with higher priority and ending with elements with lower priority. 

Anyway, I remember that in a discussion with @wojciechczerniak there were also remarks about the name. If you both agree about using another phrase for the `IndexMap` and elements within it I will change it.

@wojciechczerniak What is your opinion?

Reply:
I agree that the name does not feel right. But I don't have a better one. 

It's hard to go beyond programming, we're programming here 😄 
Queue has a definition: https://www.hackerearth.com/practice/data-structures/queues/basics-of-queues/tutorial/

"Queued", "Ordered", "Sparse", "Dense", "Indexed", "Weighted" 🤔 

I think that the closest to what we have would be a Linked Hash Map? Physical indexes to value are a hash map, but we have an additional information about their order -> links. 
I found this that looks exactly what we did: 
![hash_maps](https://user-images.githubusercontent.com/4718998/95028786-24193600-06a3-11eb-95cc-3bfa9f81cdae.png)
But maybe I'm just seeing things that proof my idea. WDYT?

Maybe `LinkedPhysicalIndexToValueMap` ?

Reply:
> I agree that the name does not feel right. But I don't have a better one.

We create something more abstract than just the regular data structure. Just some of the values are ordered/queued/whatever. I think whether any of names represent that fact 🤔 

> It's hard to go beyond programming, we're programming here 😄
> Queue has a definition: [hackerearth.com/practice/data-structures/queues/basics-of-queues/tutorial](https://www.hackerearth.com/practice/data-structures/queues/basics-of-queues/tutorial/)

I've mentioned just that as the `queue` is quite a good word for ordered elements from `0` to `n`. Literally, it sounds good for me. But I haven't focused too much on it, let's put it aside. Following the example of the **priority queue**, please take a look at the definition:

In computer science, a priority queue is an abstract data type similar to regular queue or stack data structure in which each element additionally has a "priority" associated with it. **In a priority queue, an element with high priority is served before an element with low priority**. In some implementations, if two elements have the same priority, they are served according to the order in which they were enqueued, while in other implementations, ordering of elements with the same priority is undefined.

While priority queues are often implemented with heaps, they are conceptually distinct from heaps. **A priority queue is a concept like "a list" or "a map"**; just as a list can be implemented with a linked list or an array, a priority queue can be implemented with a heap or a variety of other methods such as an unordered array.

> I think that the closest to what we have would be a Linked Hash Map? Physical indexes to value are a hash map, but we have an additional information about their order -> links.
> I found this that looks exactly what we did:
> ![hash_maps](https://user-images.githubusercontent.com/4718998/95028786-24193600-06a3-11eb-95cc-3bfa9f81cdae.png)
> But maybe I'm just seeing things that proof my idea. WDYT?
> 
> Maybe `LinkedPhysicalIndexToValueMap` ?

It sounds reasonable for me, WDYT @budnix?

Reply:
> It sounds reasonable for me, WDYT @budnix?

I would choose the `OrderedPhysicalIndexToValueMap` name 😉  The `LinkedPhysicalIndexToValueMap ` name at first glance doesn't say anything about the business logic of that specific IndexMap class. However, after looking into the source code I am able to find a connotation with the `linked` name. Summarizing if you don't feel that the `Ordered...` class name doesn't describe the module behavior enough I'm open to @wojciechczerniak suggestion and we can use `Linked...` 👍